kernel/kernel-defaults/src/main/java/io/delta/kernel/defaults/internal/expressions/DefaultExpressionEvaluator.java

```diff
@@ -560,6 +585,21 @@ ColumnVector visitCoalesce(ScalarExpression coalesce) {
             );
         }
 
+        @Override
+        ColumnVector visitLike(final Predicate like) {
+            List<Expression> children = like.getChildren();
+            int size = children.size();
+            ColumnVector left = visit(children.get(0));
+            ColumnVector right = visit(children.get(1));
+            ColumnVector escapeChar = (size == 3 && children.get(2)!=null) ?
```

Comment:
Here also, just process the all the children and return the List<ColumnVector> to `eval`

```
return LikeExpressionEvaluator.eval(children.stream().map(this::visit).collect(toList()))`
```